components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/utils/APIGatewayAdminClientTest.java

```diff
@@ -0,0 +1,608 @@
+        environment = new Environment();
+        environment.setName(ENV_NAME);
+        environment.setPassword(PASSWORD);
+        environment.setUserName(USERNAME);
+        environment.setServerURL(SERVER_URL);
+        apiGatewayAdminStub = Mockito.mock(APIGatewayAdminStub.class);
+
+        Options options = new Options();
+        ServiceContext serviceContext = new ServiceContext();
+        OperationContext operationContext = Mockito.mock(OperationContext.class);
+        serviceContext.setProperty(HTTPConstants.COOKIE_STRING, "");
+        ServiceClient serviceClient = Mockito.mock(ServiceClient.class);
+        AuthenticationAdminStub authAdminStub = Mockito.mock(AuthenticationAdminStub.class);
+        Mockito.doReturn(true).when(authAdminStub).login(Mockito.anyString(), Mockito.anyString(), Mockito.anyString());
+        Mockito.when(authAdminStub._getServiceClient()).thenReturn(serviceClient);
+        Mockito.when(serviceClient.getLastOperationContext()).thenReturn(operationContext);
+        Mockito.when(operationContext.getServiceContext()).thenReturn(serviceContext);
+        Mockito.when(apiGatewayAdminStub._getServiceClient()).thenReturn(serviceClient);
+        Mockito.when(serviceClient.getOptions()).thenReturn(options);
+        PowerMockito.whenNew(AuthenticationAdminStub.class)
+                .withArguments(Mockito.any(ConfigurationContext.class), Mockito.anyString()).thenReturn(authAdminStub);
+    }
+
+    @Test
+    public void testAPIGatewayAdminClient() throws Exception {
+        PowerMockito.whenNew(APIGatewayAdminStub.class)
+                .withArguments(Mockito.any(ConfigurationContext.class), Mockito.anyString())
+                .thenReturn(apiGatewayAdminStub);
+        APIGatewayAdminClient client = new APIGatewayAdminClient(null, environment);
+    }
+
+    @Test(expected = AxisFault.class)
+    public void testAPIGatewayAdminClientException() throws Exception {
+        PowerMockito.whenNew(APIGatewayAdminStub.class)
+                .withArguments(Mockito.any(ConfigurationContext.class), Mockito.anyString()).thenThrow(AxisFault.class);
+        APIGatewayAdminClient client = new APIGatewayAdminClient(null, environment);
+    }
+
+    @Test(expected = APIManagementException.class)
+    public void testSetSecureVaultPropertyException() throws Exception {
+        PowerMockito.whenNew(APIGatewayAdminStub.class)
+                .withArguments(Mockito.any(ConfigurationContext.class), Mockito.anyString())
+                .thenReturn(apiGatewayAdminStub);
+        Mockito.when(apiGatewayAdminStub.doEncryption(Mockito.anyString(), Mockito.anyString(), Mockito.anyString())).
+                thenThrow(RemoteException.class);
+        APIGatewayAdminClient client = new APIGatewayAdminClient(null, environment);
+        APIIdentifier identifier = new APIIdentifier("P1_API1_v1.0.0");
+        API api = new API(identifier);
+        client.setSecureVaultProperty(api, null);
```

Comment:
Better to verify or Assert a result after the test. Check this in all the tests

Reply:
Since this is an expected exception we are checking, we do not need to assert result.